cou/apps/base.py

```diff
@@ -607,6 +609,82 @@ def _get_charmhub_migration_step(self, target: OpenStackRelease) -> PreUpgradeSt
             ),
         )
 
+    def get_run_deferred_hooks_and_restart_pre_upgrade_step(self) -> list[PreUpgradeStep]:
```

Comment:
**suggestion:** 

If `get_run_deferred_hooks_and_restart_pre_upgrade_step` and `get_run_deferred_hooks_and_restart_post_upgrade_step` will only be used by anxiliary applications. Why we don't put this function in `AuxiliaryApplication` class?

Reply:
see https://github.com/canonical/charmed-openstack-upgrader/pull/504

Reply:
Resolving: addressed in the previous PR